packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -753,9 +756,38 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
       // normalize and rewrite accepted urls
       const normalizedAcceptedUrls = new Set<string>()
       for (const { url, start, end } of acceptedUrls) {
-        const [normalized] = await moduleGraph.resolveUrl(toAbsoluteUrl(url))
+        let [normalized, resolvedId] = await normalizeUrl(url, start)
```

Comment:
I have not taken a deeper look, but my guess is that this `normalizeUrl` throws an error when it fails to resolve, which is why the fallback isn't working.

Reply:
normalizeUrl was returning  the url as resolveid when it failed to resolve. I changed it to return null if it fails to resolve

Reply:
It seems we also need to `catch` the error thrown by `normalizeUrl`.
https://github.com/vitejs/vite-ecosystem-ci/actions/runs/12405630847/job/34632833233#step:8:254

Reply:
Thanks! rakkas and vitepress now passes. It seems nuxt is failing with a HMR test. It could be related to this PR.

Reply:
i could only reproduce it once locally. other times it works

Reply:
and looking at the code vuxt only uses `hot.accept(() => ...)` without deps. so this should not affect their hmr

Reply:
Ah, re-running it again worked 👍